packages/components/src/internal/components/assay/actions.ts

```diff
@@ -243,6 +244,32 @@ export function deleteAssayRuns(
     });
 }
 
+export function getImportItemsForAssayDefinitionsQM(
+    assayStateModel: AssayStateModel,
+    sampleModel?: QueryModel,
+    providerType?: string
+): OrderedMap<AssayDefinitionModel, string> {
+    let targetSQ;
+    const selectionKey = sampleModel ? sampleModel.id : undefined;
```

Comment:
In the other version of `getImportItemsForAssayDefinitions` which uses the `QueryGridModel`, this uses the `sampleModel.selectionKey`. Is that `selectionKey` actually the `id` for the QueryModel case? It looks like the `QueryGridModel` creates the selectionKey based on the keyValue and schemaQuery...ah maybe this case we know that there isn't a keyValue for the sampleModel or something like that?

Reply:
Yes, selection key is the QueryModel ID.

Reply:
selelectionKey would make no sense if there is a keyValue because that indicates that you are only looking at a single row, which we never render as a grid, and we never let you make selections on those models.

Reply:
Gotcha, thanks for clarifying.